pkg/image/api/types.go

```diff
@@ -60,6 +77,30 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation. IOW a time of the last update to ValidationResult.
+	LastValidation *unversioned.Time
+	// Result of validation performed at LastValidation. Filled by server.
+	ValidationResult SignatureValidationResult
+
+	// Following metadata fields will be set by server if the signature content is successfully parsed.
+
+	// Digest of manifest of the signed image (e.g. "sha256:xxxxx...").
+	SignedImageName string
```

Comment:
> Does Match apply when the hash matches image's parent?

That can’t happen; Docker containers do not carry the manifests of their parents.

(We do plan to have signatures which can be recognized as signatures of base layers, but that will be done using some other mechanism than just matching the image digest.)